sympy/integrals/integrals.py

```diff
@@ -1160,8 +1159,50 @@ def _eval_integral(self, f, x, meijerg=None, risch=None, manual=None,
                 parts.append(coeff * h)
             else:
                 return None
-
-        return Add(*parts)
+            current_expr = f
+            for lim in self.limits:
+                if len(lim) == 3:
+                    var, a, b = lim
+                    is_definite = True
+                elif len(lim) == 2:
+                    var, b = lim
+                    a = None
+                    is_definite = True
+                else:
+                    var = lim[0]
+                    a = b = None
+                    is_definite = False
+                if is_definite and a is not None and b is not None:
+                    attempted = integrate(current_expr, (var, a, b), meijerg=meijerg, risch=risch,
+                                       manual=manual, heurisch=heurisch, conds=conds)
+                    if attempted == current_expr:
+                        break
+                    current_expr = attempted
```

Comment:
What is this doing? Hasn't the code above already attempted to evaluate the integral? Why is this recursively calling `integrate`?